### DISCUSSION TITLE
Breadcrumb / Categories

### DISCUSSION THREAD

**Discussion by souksou** [swellstores/community#323] [Help] [answered]
Hello guys,

I just have question about categories breadcrumb, do you know if there is a existing feature for that or how can I retrieve it for a product  directly in the API? 

Example: Category > Sub category

Exacly as the admin interface.

Thanks a lot for your help.

**Comment by logeshswell** [comment, marked as answer]
@souksou Can you check this discussion?
https://github.com/orgs/swellstores/discussions/107#discussioncomment-6179346

**Comment by Lulololu** [comment]
Hi guys, 
Hi @logeshswell 

Working on the same project as @souksou 

Got 2 questions :

- We managed to get the breadcrumbs with swell.js lib, on front end `(expand: [breadcrumb])`
Yet, with GraphQL, nothing more than the previous documentation is displayed when we load the documentation again. We do not receive any breadcrumbs at all. **Is it limited to Rest ?** If not, would we normally be able to load a new GraphQL documentation ? Or should we have to type some kind of "expand" query ?

- Beyond that, **we'd like to know if, globally, Swell is GraphQL ready**. I mean, the online documentation feels sometimes incomplete, and we would need to assess if it's worth all the efforts (because of the cool DX) for us to keep working with GraphQL.


**Thanks a lot for everything !**

**Reply by logeshswell** [reply]
> Is it limited to Rest?

Yes, I believe. We need to add support for custom `expand` in GraphQL.

> Is Swell GraphQL ready?

Yes, it is ready and being used in production. But, you may miss some features supported in swell-js but not in GraphQL at the moment. For example, custom `expand` as mentioned above.

We are working hard to support all of the possible features in graphQL as well.

**Reply by logeshswell** [reply]
If you feel our documentation lags some important details on GraphQL, please feel free to report to us.

Secondly, you can explore our GraphQL playground to have complete details on possible queries and mutations supported in Swell.
https://developers.swell.is/frontend-api/setup#playground

<img width="791" alt="Screenshot 2023-07-26 at 9 32 42 PM" src="https://github.com/swellstores/community/assets/100766131/23d882d5-547b-44d6-b61e-cc23d2d34b3a">

**Comment by Lulololu** [comment]
@logeshswell Thanks a lot for your kind assistance,

Well, sincerely, this feature is essential for us (we need it to retrieve breadcrumbs, categories hierarchy, and so on) and, if not GraphQL-ready, we're going to switch to the Rest Library.

That's an illustration of something we would have been happy to learn beforehand.

**Thanks again for your help :-)**
